heavyai/connection.py

```diff
@@ -505,6 +505,79 @@ def get_table_details(self, table_name):
         details = self._client.get_table_details(self._session, table_name)
         return _extract_column_details(details.row_desc)
 
+    def get_table_details_with_metadata(self, table_name):
```

Comment:
Can we name this `get_table_details` and the next method `get_table_details_for_database`? I believe this will be consistent with how most of the other methods are named. Also, the existing `get_table_details` method can be renamed to something like `get_table_column_details` or `get_column_details`, which would more accurately reflect what the method does.